criu/image.c

```diff
@@ -134,7 +134,12 @@ InventoryEntry *get_parent_inventory(void)
 
 	dir = openat(get_service_fd(IMG_FD_OFF), CR_PARENT_LINK, O_RDONLY);
 	if (dir == -1) {
-		pr_warn("Failed to open parent directory\n");
+		/*
+		 * If we have no parent directory (no previous dump) then we
+		 * don't have parent inventory image too, it's expected.
+		 */
+		if (errno != ENOENT)
+			pr_warn("Failed to open parent directory\n");
```

Comment:
you have to report an error if errno isn't ENOENT, don't you?

Reply:
We print the warning "Failed to open parent directory" from get_parerent_inventory() to be notified that we had some unexpected problem on open, for instance we have a parent directory but have no access. But failing here is too early, we get to detect_pid_reuse() only if we have a parent pagemap, that's the proper place to fail: we know that there is a parent pagemap but we can't access the parent inventory => can't detect pid-reuse => fail.

Reply:
This sounds reasonable. Could you add a comment with this explanation?

Reply:
Sure, done. Moved the general part before function.